### ISSUE TITLE
get_balance and other similar APIs use for multiple addresses 

### ISSUE THREAD

**Issue by gbenattar** [romanz/electrs#73]
When calling get_balance with an array containing more than one addresses (see below), the returned balance is not an aggregated sum of all addresses but the balance for the first one (all others are discarded). Not sure it is an expected behavior of electrum but the fact that it expect an array of addresses make the client expect receiving either a list of balances or an aggregated sum.

What do you think?

* get_balance for tb1qppdty6emlujdwrhdjfyy3l70qd0jeapnhhzj80
```
{"id":1,"jsonrpc":"2.0","result":{"confirmed":50000,"unconfirmed":0}}
```

* get_balance for tb1qtflk05mgferry3ejp76z4uyqvpae2r9dfueejs
```
{"id":1,"jsonrpc":"2.0","result":{"confirmed":18916559,"unconfirmed":0}}
```
* get balance for [tb1qppdty6emlujdwrhdjfyy3l70qd0jeapnhhzj80, tb1qtflk05mgferry3ejp76z4uyqvpae2r9dfueejs]
```
echo '
{  
   "id":1,
   "method":"blockchain.address.get_balance",
   "params":[  
      "tb1qppdty6emlujdwrhdjfyy3l70qd0jeapnhhzj80",
      "tb1qtflk05mgferry3ejp76z4uyqvpae2r9dfueejs"
   ]
}
' | netcat localhost 60001
```
 * Actual
```
{"id":1,"jsonrpc":"2.0","result":{"confirmed":50000,"unconfirmed":0}}
```

 * Expected
```
{"id":1,"jsonrpc":"2.0","result":{"confirmed": 18966559,"unconfirmed":0}}
```

**Comment by romanz** [comment]
Thanks for reporting the issue!

It seems that the Electrum protocol specification doesn't cover this use-case:
http://docs.electrum.org/en/latest/protocol.html#blockchain-address-get-balance

**Comment by romanz** [comment]
I think that your suggestion makes great sense, and I'd be happy to merge this change :)

**Comment by gbenattar** [comment]
This is a great use case for HD Wallet, the same applies also for getting list-unspent and transaction history but in those, who should have a way to map the address to the result: 

```
[  
   {  
      "address":"1",
      "result":{  
        [...]
      }
   },
   {  
      "address":"2",
      "result":{  
        [...]
      }
   }
]
```

**Comment by romanz** [comment]
I'd suggest to keep the old convention (returning aggregate results for all specified addresses) for backward compatibility.
If you need to group the results by address, we can think about adding a new API that supports grouping.

**Comment by gbenattar** [comment]
I agree - this is what I added in the Javascript library, creating new API which instead of [..]address[..] are named [...]addresses[..].
